resource/opened.go

```diff
@@ -7,12 +7,46 @@ package resource
 
 import (
 	"io"
+	"strings"
+
+	"github.com/juju/errors"
 )
 
+type multiError []error
+
+func (m multiError) Error() string {
+	messages := make([]string, len(m))
+	for i, err := range m {
+		messages[i] = err.Error()
+	}
+	return strings.Join(messages, ", and also ")
+}
+
+// CombineErrors converts a set of errors (which might be nil) into
+// one. If there are no errors, this returns an untyped nil, and if
+// there's one error it's passed through directly.
+func CombineErrors(errs ...error) error {
```

Comment:
Do we already have something like this somewhere? It seems like we would. Where should it live if not? github.com/juju/errors?